### DISCUSSION TITLE
Where is Cavern headed?

### DISCUSSION THREAD

**Discussion by nicoleise** [VoidXH/Cavern#119] [General]
Hi, 

First of all, incredible work and vision. Stumbled upon this on an audio forum and have looked extensively for a few months. 

I hope this is the appropriate way to ask and also a fair question to ask. I realize questions like "when is it done" are impolite, and please know I'm not asking in that sense.

Will Cavern/Cavernize/derivatives of this project **eventually** be suited to replace an AVP (AVR sans amplifiers)? Is that in the plans or scope? 

In my mind it's a interesting proposition if a PC device could decode Atmos on the fly from HDMI, disc, storage or network sources, and many of the features incorporated in Cavern seem to be very compatible with this notion. 

However, it could be that you're targeting DIYers or a "budget friendly" option, it could be you're intending on building very capable and expensive professional gear or that neither of these are on your roadmap at all. 

Just curious if I should be on the lookout for good AVPs or if my living room (currently being being built, so a few years out) would enjoy Cavern as the brains of the operation in say a year or two. 

Also from a contribution point of view. I don't really know much coding, but there are other aspects I might be able to help with, depending on the direction of the project. 

Kind regards, 
Nicolai

**Comment by VoidXH** [comment]
Hi Nicolai,

Cavern is aiming to be a complete replacement for an AVR for special cases, including decoding, rendering, and calibration. However, there are currently some limitations that will be added, and some that can't be added.

Will be added, but can take years:
- Real-time decoding of currently supported formats
- Full object-based support for AC-4 and DTS-UHD (development starts when the first contents are released)
- Support for over 8 channels

Will not be added:
- TrueHD Atmos and DTS:X support (these are not publicly documented formats, and likely will never be)
- HDMI input support (no such hardware could exist legally)

Best,
Bence
